src/main/java/ml/shifu/shifu/core/dtrain/wdl/DenseLayer.java

```diff
@@ -189,12 +189,15 @@ public void setL2reg(float l2reg) {
         // gradients compute and L2 reg here
         for(int i = 0; i < this.in; i++) {
             for(int j = 0; j < this.out; j++) {
-                this.wGrads[i][j] += (this.lastInput[i] * backInputs[j] * sig); // basic derivatives
-                this.wGrads[i][j] += (this.l2reg * this.weights[i][j] * sig);// l2 loss derivatives
+                // basic derivatives
+                this.wGrads[i][j] += (this.lastInput[i] * backInputs[j] * sig);
+                // l2 loss derivatives
+                this.wGrads[i][j] += (this.l2reg * this.weights[i][j] * sig);
```

Comment:
Just noticed such two lines, seems we can save one mulitply operation by merging together